### PR TITLE
ESConnection backward config compatibility

### PR DESCRIPTION
**Pull request by language-shprt**

### CODE REVIEW

File: bspump/elasticsearch/connection.py [hunk 1]
```diff
@@ -47,11 +47,8 @@ def __init__(self, connection, index, max_size):
 		self.FailLogMaxSize = connection.FailLogMaxSize
 		self.FilterPath = connection.FilterPath
 
-		# Get credentials
-		username = connection.Config.get('username')
-		password = connection.Config.get('password')
-		api_key = connection.Config.get('api_key')
-		url = connection.Config.get('url')
+		# Get credentials for authorization
+		username, password, api_key, url = get_credentials_from_config(self.Config)
```
Comment (mejroslav) [inline bspump/elasticsearch/connection.py:51]: `ElasticSearchBulk` does not have attribute `Config`. Use `connection.Config`.

File: bspump/elasticsearch/connection.py [hunk 2]
```diff
@@ -607,9 +600,31 @@ def build_headers(username, password, api_key):
...
+	# new configuration
+	username = configuration.get('username')
+	password = configuration.get('password')
+	api_key = configuration.get('api_key')
+	url = configuration.get('url')
+
+	# backward compatibility with the format http://<username>:<password>@path
+	auth_credentials = configuration.get('elasticsearch_url')
+	if auth_credentials is not None:
+		asab.LogObsolete.warning("The configuration of ESConnection is outdated, please update the [connection:] section.")
```
Comment (mejroslav) [inline bspump/elasticsearch/connection.py:623]: Add `struct_data: {"eof": "..."}` and provide more detailed description of what is wrong with the configuration.
The section is not named `[connection:]`.

File: bspump/elasticsearch/connection.py [hunk 3]
```diff
@@ -607,9 +600,31 @@ def build_headers(username, password, api_key):
...
+def get_credentials_from_config(configuration):
+	"""
+	Parses configurations to return credentials
+	"""
+
+	# new configuration
+	username = configuration.get('username')
+	password = configuration.get('password')
+	api_key = configuration.get('api_key')
+	url = configuration.get('url')
```
Comment (mejroslav) [inline bspump/elasticsearch/connection.py:618]: What if there are multiple URLs? Do you split them and handle them separately?

File: bspump/elasticsearch/connection.py [hunk 4]
```diff
@@ -607,9 +600,31 @@ def build_headers(username, password, api_key):
...
+
+	# backward compatibility with the format http://<username>:<password>@path
+	auth_credentials = configuration.get('elasticsearch_url')
+	if auth_credentials is not None:
+		asab.LogObsolete.warning("The configuration of ESConnection is outdated, please update the [connection:] section.")
+		pattern = re.compile(r"(https?://)((.*):(.*)@)?([^#]*)$")
+		path_split = pattern.findall(auth_credentials)[0]
+		L.debug(path_split)
+		url_scheme, user_info, username, password, url_path = path_split
```
Comment (mejroslav) [inline bspump/elasticsearch/connection.py:627]: You are getting these information from `Config["elasticsearch_url"]`. But what if someone specifies it in `Config["url"]`?

**Review by mejroslav — CHANGES_REQUESTED** [review]
Do not forget to test it somewhere.

File: bspump/elasticsearch/connection.py (hunk 3 above)
Comment (language-shprt) [reply bspump/elasticsearch/connection.py:618]: That's handled in the code.

File: bspump/elasticsearch/connection.py (hunk 4 above)
Comment (language-shprt) [reply bspump/elasticsearch/connection.py:627]: In this case can I please get a better input on old config options please? Isn't it this format for "old" configuration format: http://<username>:<password>@path? Also, Config["url"] is handled in the first part of the function or I probably don't get what you mean?

File: bspump/elasticsearch/connection.py (hunk 1 above)
Comment (language-shprt) [reply bspump/elasticsearch/connection.py:51]: ok, thanks

File: bspump/elasticsearch/connection.py (hunk 4 above)
Comment (eliska-n) [reply bspump/elasticsearch/connection.py:627]: I bet the "elasticsearch_url" option has never been in bspump (it is an option in asab:storage section) and it was only typo in the task description. Elena, please use real site configs as a reference if you're not sure.

File: bspump/elasticsearch/connection.py [hunk 5]
```diff
@@ -607,9 +592,53 @@ def build_headers(username, password, api_key):
...
+	password = configuration.get('password')
+	api_key = configuration.get('api_key')
+	config_url = configuration.get('url')
+
+	if "@" not in config_url:
+		# new configuration format
+		node_urls = split_multiple_nodes_url(config_url)
+	else:
+		# outdated configuration format http://<username>:<password>@path
+		asab.LogObsolete.warning("The configuration format http://<username>:<password>@path is outdated, please update [connection:ESConnection] section.")
```
Comment (eliska-n) [inline bspump/elasticsearch/connection.py:616]: write what is the right format

File: bspump/elasticsearch/connection.py [hunk 6]
```diff
@@ -47,18 +47,15 @@ def __init__(self, connection, index, max_size):
...
+		# Get credentials for authorization and nodes' urls
+		username, password, api_key, node_urls = get_credentials_from_config(connection.Config)
 
 		# Build headers
 		self.Headers, self._auth = build_headers(username, password, api_key)
 
 		# Build ssl context
 		self.SSLContextBuilder = SSLContextBuilder('connection:{}'.format(id))
-		if url.startswith('https://'):
+		if node_urls[0].startswith('https://'):
```
Comment (eliska-n) [inline bspump/elasticsearch/connection.py:58]: if any from the urls

File: bspump/elasticsearch/connection.py [hunk 7]
```diff
@@ -47,18 +47,15 @@ def __init__(self, connection, index, max_size):
 		self.FailLogMaxSize = connection.FailLogMaxSize
 		self.FilterPath = connection.FilterPath
 
-		# Get credentials
-		username = connection.Config.get('username')
-		password = connection.Config.get('password')
-		api_key = connection.Config.get('api_key')
-		url = connection.Config.get('url')
+		# Get credentials for authorization and nodes' urls
+		username, password, api_key, node_urls = get_credentials_from_config(connection.Config)
```
Comment (eliska-n) [inline bspump/elasticsearch/connection.py:51]: I don't understand why this "Bulk" object is reading the configuration in the first place

**Comment by ateska** [comment]
This has to be realigned with https://github.com/LibertyAces/BitSwanPump/pull/184

Rebase this please.

**Comment by eliska-n** [comment]
This is not needed. #184 is sufficient.

**Comment by language-shprt** [comment]
Since nobody wants this anymore...

**Comment by language-shprt** [comment]
Not needed.

**Pull request closed without merging**